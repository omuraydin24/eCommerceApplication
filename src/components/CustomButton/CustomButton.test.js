import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import Chance from 'chance';
import Button from '.';
import buttonStyles from './styles';

const chance = new Chance();

test('is match with snapshot', () => {
  const component = render(<Button />);
  expect(component).toMatchSnapshot();
});

test('should display title if given', () => {
  const testText = chance.string();
  const component = render(<Button title={testText} />);

  const text = component.getByTestId('button-title').children[0];

  expect(text).toBe(testText);
});

test('should rendered with given style', () => {
  const testType = 'outline';
  const component = render(<Button type={testType} />);

  const componentStyle = component.getByTestId('button-view-container').props
    .style;

  expect(componentStyle).toMatchObject(buttonStyles[testType].container);
});

test('should not triggered when loading prop is given', () => {
  const mockFn = jest.fn();
  const component = render(<Button loading onPress={mockFn} />);

  const pressable = component.getByTestId('button-view-container');

  fireEvent.press(pressable);

  expect(mockFn).not.toBeCalled();
});

test('should not triggered when disable prop is given', () => {
  const mockFn = jest.fn();
  const component = render(<Button disable onPress={mockFn} />);

  const pressable = component.getByTestId('button-view-container');

  fireEvent.press(pressable);

  expect(mockFn).not.toBeCalled();
});

test('should set disabled style when disable prop is given', () => {
  const component = render(<Button disable />);

  const componentStyle = component.getByTestId('button-view-container').props
    .style;

  expect(componentStyle).toMatchObject(buttonStyles.disabled.container);
});
