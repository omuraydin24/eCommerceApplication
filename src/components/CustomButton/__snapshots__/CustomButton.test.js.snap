// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is match with snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#2A59FE",
      "borderRadius": 4,
      "height": 40,
      "justifyContent": "center",
      "margin": 10,
      "marginHorizontal": 15,
      "opacity": 1,
    }
  }
  testID="button-view-container"
/>
`;
